### ISSUE TITLE
Unable to connect to Kusto

### ISSUE THREAD

**Issue by TylerLeonhardt** [dotnet/interactive#2799]
### Describe the bug
Please provide as much information as you can.

### Please complete the following:

**Which version of .NET Interactive are you using?** (In a notebook, run the `#!about` magic command. ):

- OS
    - [ ] Windows 11
    - [ ] Windows 10
    - [x] macOS
    - [ ] Linux (Please specify distro)
    - [ ] iOS
    - [ ] Android
 - Browser 
    - [x] Chrome
    - [ ] Edge
    - [ ] Firefox
    - [ ] Safari
 - Frontend
    - [ ] Jupyter Notebook
    - [ ] Jupyter Lab
    - [ ] nteract
    - [ ] Visual Studio Code 
    - [x] Visual Studio Code Insiders
    - [ ] Visual Studio
    - [ ] Other (please specify)

### Screenshots
![image](https://user-images.githubusercontent.com/2644648/223200725-a6a7a063-30da-4c90-a846-7df06e2107ab.png)
(removed private data)

**Comment by jonsequitur** [comment]
What version of VS Code and Polyglot Notebooks are you using?

**Comment by TylerLeonhardt** [comment]
Polyglot: v1.0.4153011

VS Code stuff:

Version: 1.77.0-insider (Universal)
Commit: 0cab3cc3cbe4d4ac2cbf30e8c30351d9db6dcfb4
Date: 2023-03-06T05:21:07.151Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin arm64 22.3.0
Sandboxed: Yes

**Comment by brettfo** [comment]
@TylerLeonhardt The Kusto connector installs a dotnet global tool to do its work, can you run this `dotnet tool list --global` and check the version of `microsoft.sqlserver.kustoservicelayer.tool`?  I'm hoping to see `1.2.0`.  Similarly for `microsoft.sqlserver.sqltoolsservicelayer.tool`.

**Comment by TylerLeonhardt** [comment]
I see just:

```
microsoft.sqlserver.kustoservicelayer.tool      1.2.0        MicrosoftKustoServiceLayer
```

**Comment by TylerLeonhardt** [comment]
![Image](https://user-images.githubusercontent.com/2644648/223227514-addcabb6-69ba-472b-b549-fc6d6bd39827.png)

looks like it in my terminal in vscode.

**Comment by corivera** [comment]
There have been other issues with the tools not showing up properly on the PATH. I'll add the full Kusto tool path to the services and see if that resolves the issue.

**Comment by TylerLeonhardt** [comment]
@jonsequitur and I sat down, the fix, for me, was the set my `Environment.CurrentDirectory` to the dotnet tools directory and things started working.

**Comment by corivera** [comment]
Is $env:USERPROFILE/.dotnet/tools present in the PATH on that machine?

**Comment by jonsequitur** [comment]
`$USERPROFILE/.dotnet/tools`, no. `$USER/.dotnet/tools`, yes.

**Comment by corivera** [comment]
What does $USER resolve to here? It's not defined on my machine

**Comment by jonsequitur** [comment]
Yeah, that `Win32Exception` is always a fun vestigial name to find on Linux and Mac.

**Comment by corivera** [comment]
So the tools directory is in the PATH, but the tool can only be found if you're working in that tools folder?

**Comment by jonsequitur** [comment]
The tools directory is in the path but the extension isn't calculating the path correctly on macOS. It expands the unknown `$USERPROFILE` variable as an empty string, resulting in the extension looking for `MicrosoftKustoServiceLayer` in the current directory (which was the Downloads directory in @TylerLeonhardt's screen shot). This is why changing the working directory to the tools directory was an effective workaround.

**Comment by corivera** [comment]
What specific part do you mean by "extension" here?

**Comment by jonsequitur** [comment]
`Microsoft.DotNet.Interactive.Kql`. Presumably `Microsoft.DotNet.Interactive.SqlServer` has the same problem.

**Comment by corivera** [comment]
We aren't using USERPROFILE in those, though. We just use the tool name as the path for the process to start. So it currently would rely entirely on what's in the PATH (unless I'm misunderstanding something here). This is where we pass the service path to the KQL command. It gets passed through, unmodified, to ProcessStartInfo eventually in ToolsServiceClient. https://github.com/dotnet/interactive/blob/main/src/Microsoft.DotNet.Interactive.Kql/KqlKernelExtension.cs#L20
